drivers/gpio/gpio_mcux.c

```diff
@@ -292,8 +292,8 @@ static uint32_t get_port_pcr_irqc_value_from_flags(const struct device *dev, uin
 #endif /* !defined(FSL_FEATURE_PORT_HAS_NO_INTERRUPT) && FSL_FEATURE_PORT_HAS_NO_INTERRUPT */
 #endif /* !(defined(CONFIG_PINCTRL_NXP_IOCON)) */
 
-#if (defined(FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT) && \
-		FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT)
+#if (defined(FSL_FEATURE_PORT_HAS_NO_INTERRUPT) && FSL_FEATURE_PORT_HAS_NO_INTERRUPT) || \
+		(!defined(FSL_FEATURE_SOC_PORT_COUNT))
```

Comment:
the logic seems not correct. use to be FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT but now FSL_FEATURE_PORT_HAS_NO_INTERRUP

Reply:
Hi hake, as my comment in this PR,  cannot use FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT to judge if ICR register exist. This feature only use to judge if bitfield IRQS exist in ICR register. So suggest to use feature FSL_FEATURE_PORT_HAS_NO_INTERRUPT to do this judgment. This judgment is also used in our sdk, it can work fine. the comment you mentioned that this may impact other platform need to be considered, I tested the boards(frdmmcxn947,frdmmcxn236,frdmmcxa156,frdmmcxa153) I have, all can work fine. Could you help do some tests on other platform?

Reply:
please take a look at https://github.com/zephyrproject-rtos/zephyr/pull/88979. this avoid such potential issue